PSix/OnboardingViewController.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  OnboardingViewController.swift
+//  PSix
+//
+//  Created by Turzion, Avihu on 4/13/15.
+//  Copyright (c) 2015 PayPal. All rights reserved.
+//
+
+import UIKit
+
+class OnboardingViewController: UIViewController {
+    
+    enum LoginMode {
+        case Login, Logout, DuringLogin
+    }
+    
+    private let LOGIN_BUTTON_TITLE = "Login with Facebook"
+    private let LOGOUT_BUTTON_TITLE = "Sign out"
+    private let LOGGING_IN_BUTTON_TITLE = "Logging in..."
```

Comment:
cool trick. but you can add a method in the enum that returns the button title based on the mode

```swift
enum State {
  func buttonTitle(){
    switch self {
      case .Login: return "Login with Facebook"
    }
  }
}
```